From: Markus Koschany <apo@debian.org>
Date: Wed, 27 Jul 2016 11:00:08 +0200
Subject: revert switch to jprante's jflex plugin

---
 build.gradle       |  3 ++-
 smali/build.gradle | 12 +++++++++---
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/build.gradle b/build.gradle
index 32675b8..eb445e5 100644
--- a/build.gradle
+++ b/build.gradle
@@ -105,7 +105,8 @@ subprojects {
                 antlr: 'org.antlr:antlr:3.5.2',
                 stringtemplate: 'org.antlr:stringtemplate:3.2.1',
                 commons_cli: 'commons-cli:commons-cli:1.2',
-                jflex_plugin: 'org.xbib.gradle.plugin:gradle-plugin-jflex:1.1.0',
+                jflex: 'de.jflex:jflex:1.6.1',
+                jflex_plugin: 'co.tomlee.gradle.plugins:gradle-jflex-plugin:0.0.2',
                 proguard_gradle: 'net.sf.proguard:proguard-gradle:5.2.1',
                 dx: 'com.google.android.tools:dx:1.7',
                 gson: 'com.google.code.gson:gson:2.3.1'
diff --git a/smali/build.gradle b/smali/build.gradle
index 318b5a9..a892399 100644
--- a/smali/build.gradle
+++ b/smali/build.gradle
@@ -30,7 +30,7 @@
  */
 
 apply plugin: 'antlr'
-apply plugin: 'org.xbib.gradle.plugin.jflex'
+apply plugin: 'jflex'
 
 buildscript {
     repositories {
@@ -46,8 +46,13 @@ configurations {
     // Remove the full antlr library that's added by the antlr plugin. We manually
     // add the smaller antlr_runtime library instead
     compile.exclude group: 'org.antlr', module: 'antlr'
+
+    // The jflex lexer doesn't have any runtime dependencies, so remove the dependency
+    // that gets added by the jflex plugin
+    compile.exclude group: 'de.jflex', module: 'jflex'
 }
 
+
 sourceSets {
     main {
         resources {
@@ -82,6 +87,7 @@ dependencies {
     testCompile depends.junit
 
     antlr depends.antlr
+    jflex depends.jflex
 }
 
 processResources.inputs.property('version', version)
@@ -114,8 +120,8 @@ generateGrammarSource {
     outputDirectory = new File(outputDirectory, 'org/jf/smali')
 }
 
-jflex {
-    generateDir = new File(generateDir, 'org/jf/smali')
+generateJFlexSource {
+    outputDirectory = new File(outputDirectory, 'org/jf/smali')
 }
 
 uploadArchives {
